### PR TITLE
MGMT-13580: GC orphan hosts

### PR DESCRIPTION
**Pull request by slaviered**
## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/host.go [hunk 1]
```diff
@@ -1379,6 +1380,22 @@ func (m *Manager) ResetHostValidation(ctx context.Context, hostID, infraEnvID st
 	}
 }
 
+func (m Manager) DeleteOrphanHosts(ctx context.Context) error {
+	var hosts []*models.Host
+
+	db := m.db.Unscoped()
+	reply := db.Model(&models.Host{}).
+		Where("NOT EXISTS (SELECT 1 FROM infra_envs WHERE infra_envs.id = infra_env_id)").
+		Delete(&hosts)
```
Comment (ori-amizur) [inline internal/host/host.go:1389]:
```suggestion
	reply := db.Where("NOT EXISTS (SELECT 1 FROM infra_envs WHERE infra_envs.id = infra_env_id)").
		Delete(&models.Host{})
```

File: internal/host/host.go [hunk 2]
```diff
@@ -1379,6 +1380,22 @@ func (m *Manager) ResetHostValidation(ctx context.Context, hostID, infraEnvID st
 	}
 }
 
+func (m Manager) DeleteOrphanHosts(ctx context.Context) error {
+	var hosts []*models.Host
```
Comment (ori-amizur) [inline internal/host/host.go:1384]: Not needed

File: internal/host/host.go (hunk 1 above)
Comment (slaviered) [reply internal/host/host.go]: Fixed

File: internal/host/host.go (hunk 2 above)
Comment (slaviered) [reply internal/host/host.go]: Fixed

**Review by ori-amizur — APPROVED** [review]

**Pull request merged**